packages/discovery-react-components/src/components/TooltipHighlight/TooltipHighlight.tsx

```diff
@@ -0,0 +1,184 @@
+    isOpen: false
+  });
+
+  useEffect(() => {
+    // Compute parameters to overlay the tooltip over the active item
+    const highlightDivRect = parentDiv.current?.getBoundingClientRect() || new DOMRect();
+    const isOpen = tooltipAction.tooltipEvent !== TooltipEvent.LEAVE;
+    const clickRect = tooltipAction.rectActiveElement || new DOMRect();
+    const tooltipRect = new DOMRect(
+      clickRect.x - highlightDivRect.x,
+      clickRect.y - highlightDivRect.y,
+      clickRect?.width,
+      clickRect?.height
+    );
+    const tooltipUpdate = {
+      rectTooltipArea: tooltipRect,
+      tooltipContent: tooltipAction.tooltipContent || <div></div>,
+      isOpen: !!tooltipAction.tooltipContent && isOpen
+    };
+    setTooltipInfo(tooltipUpdate);
+  }, [tooltipAction, setTooltipInfo, parentDiv]);
+
+  return (
+    // Outter div is required to provide tooltip element with position information
+    // "pointerEvents" = "none" so that underlying elements can react to mouse events
+    <div
+      className={cx(baseTooltipPlaceContent)}
+      style={{
+        top: tooltipInfo.rectTooltipArea.y,
+        left: tooltipInfo.rectTooltipArea.x
+      }}
+    >
+      <Tooltip
+        autoOrientation={true}
+        tabIndex={0}
+        showIcon={false}
+        open={tooltipInfo.isOpen}
+        triggerText={
+          <div
+            style={{
+              // Provide size so that tooltip knows boundry where to draw the tooltip
+              width: tooltipInfo.rectTooltipArea.width,
+              height: tooltipInfo.rectTooltipArea.height,
+              pointerEvents: 'none'
+            }}
+          />
+        }
+        children={<div>{tooltipInfo.tooltipContent}</div>}
+      />
+    </div>
+  );
+};
+
+export function calcToolTipContent(
+  facetInfoMap: FacetInfoMap,
+  facetId: string,
+  enrichValue: string
+) {
+  const tableContent = [];
+  let enrichColor = '';
+  let enrichFacetDisplayname = '';
+  if (facetInfoMap[facetId]) {
+    enrichColor = facetInfoMap[facetId].color;
+    enrichFacetDisplayname = facetInfoMap[facetId].displayName;
+    if (
+      enrichFacetDisplayname.localeCompare(enrichValue, undefined, { sensitivity: 'base' }) == 0
+    ) {
+      // This case applies to keyowrds
+      enrichFacetDisplayname = 'Keyword';
+    }
+    // Will have multiple entries after overlapping is implemented
+    tableContent.push({
+      enrichColor: enrichColor,
+      enrichFacetDisplayname: ellipsisMiddle(enrichFacetDisplayname),
+      enrichValue: ellipsisMiddle(enrichValue)
+    });
+  }
+
+  // sample data for development
+  // tableContent.push({
+  //   enrichColor: 'red',
+  //   enrichFacetDisplayname: 'short',
+  //   enrichValue: ellipsisMiddle('extra super duper long long 44 characters')
+  // });
+  // tableContent.push({
+  //   enrichColor: 'green',
+  //   enrichFacetDisplayname: ellipsisMiddle('extra super duper long long 44 characters'),
+  //   enrichValue: 'short'
+  // });
```

Comment:
reminder to delete commented-out code before merging

Reply:
done